dev-tools/xtask/src/a4x2_package.rs

```diff
@@ -0,0 +1,382 @@
+            .arg(&format!("curl -sSfL --retry 10 https://get.nexte.st/'{}'/illumos | gunzip | tar -xvf - -C ~/.cargo/bin", nextest_config.nextest_version.recommended))
+            .run()?;
+        env.home_dir.join(".cargo/bin/cargo-nextest")
+    } else {
+        // XXX do we need to test nextest exists & is the right version? need to
+        // see if the incidental errors from stuff below already take care of
+        // telling someone outside of CI what they need to be doing.
+        Utf8PathBuf::from(cmd!(sh, "/usr/bin/which cargo-nextest").read()?)
+    };
+
+    // Build and generate the live tests bundle
+    // generates
+    // - target/live-tests-archive.tgz
+    let cargo = &env.cargo;
+    cmd!(sh, "{cargo} xtask live-tests").run()?;
+
+    let live_test_bundle_dir = env.work_dir.join("live-tests-bundle");
+    sh.create_dir(&live_test_bundle_dir)?;
+
+    // a tar within a tar, a door behind a door
+    cmd!(
+        sh,
+        "tar -xzf target/live-tests-archive.tgz -C {live_test_bundle_dir}"
+    )
+    .run()?;
+
+    // and we need nextest to execute it
+    sh.copy_file(&nextest_path, &live_test_bundle_dir)?;
+
+    // Finally, the script that will run nextest
+    // TODO: should we make this into a rust program that runs in the VM?
+    let switch_zone_script_path = live_test_bundle_dir.join("run-live-tests");
+    let switch_zone_script = r#"
+        set -euxo pipefail
+        /opt/oxide/omdb/bin/omdb -w nexus blueprints target enable current
+
+        TMPDIR=/var/tmp ./cargo-nextest nextest run \
+            --profile=live-tests \
+            --archive-file live-tests-archive/omicron-live-tests.tar.zst \
+            --workspace-remap live-tests-archive
+    "#;
+    sh.write_file(&switch_zone_script_path, switch_zone_script)?;
+    cmd!(sh, "chmod +x {switch_zone_script_path}").run()?;
+
+    // output tar will have live-tests and nextest. this will get uploaded
+    // into one of the a4x2 sleds, which is why it is a self contained tar
+    let out_dir = &env.out_dir;
+
+    // intentionally relative path so the tarball gets relative paths
+    let _popdir = sh.push_dir(&env.work_dir);
+    let bundle_dir_name = live_test_bundle_dir.file_name().unwrap();
+    cmd!(sh, "tar -czf {out_dir}/live-tests-bundle.tgz {bundle_dir_name}/")
+        .run()?;
+
+    Ok(())
+}
+
+fn build_a4x2(sh: &Shell, env: &Environment) -> Result<()> {
+    cmd!(sh, "banner testbed").run()?;
+    let testbed_dir = env.work_dir.join("testbed");
+    let a4x2_dir = testbed_dir.join("a4x2");
+
+    // TODO move this clone & build to prepare, so that if the omicron build
+    // takes awhile we wont have lost our git token when we get here. Relevant
+    // for CI but not for running locally.
+    cmd!(
+        sh,
+        "git clone https://github.com/oxidecomputer/testbed {testbed_dir}"
+    )
+    .run()?;
+
+    // XXX use main branch
+    // cmd!(sh, "git -C {testbed_dir} checkout artemis/init-in-smf").run()?;
```

Comment:
Could make this a command-line argument.  It could accept a branch name or alternatively a local path to a clone if you want to do anything other than "main".